language/move-vm/types/src/native_functions/context.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    interpreter_context::InterpreterContext,
+    loaded_data::{runtime_types::Type, types::FatType},
+    values::Struct,
+};
+use libra_types::{account_address::AccountAddress, language_storage::ModuleId};
+use move_core_types::{gas_schedule::CostTable, identifier::IdentStr};
+use std::fmt::Write;
+use vm::errors::VMResult;
+
+/// Native function context.
+pub trait NativeContext {
+    /// Gets caller module.
+    fn caller_module(&self) -> Option<&ModuleId>;
+    /// Gets current function module.
+    fn self_module(&self) -> &ModuleId;
+    /// Gets interpreter ref.
```

Comment:
I got it.  

> Just leaving a comment for now with some questions for you.
> We'll be having a meeting about this tomorrow because few people in the team have thought about native functions and had ideas on how to move forward.
> We will be in a better spot then to suggest specific changes.
> 
> Overall a pointer to (or just an explanation on) how you intend to use it may help us understand certain API (as explained in comments).
> In other words we recognize that plugging native functions is pretty vital and it would be great to understand the different use cases.
> 
> Good work btw!

Thank you for your review.